programs/marginfi/src/instructions/marginfi_account/withdraw.rs

```diff
@@ -131,17 +145,20 @@ pub fn lending_account_withdraw<'info>(
 
     marginfi_account.lending_account.sort_balances();
 
-    // Check account health, if below threshold fail transaction
-    // Assuming `ctx.remaining_accounts` holds only oracle accounts
-    let (risk_result, _engine) = RiskEngine::check_account_init_health(
-        &marginfi_account,
-        ctx.remaining_accounts,
-        &mut Some(&mut health_cache),
-    );
-    risk_result?;
-    health_cache.program_version = PROGRAM_VERSION;
-    health_cache.set_engine_ok(true);
-    marginfi_account.health_cache = health_cache;
+    // Note: during liquidating, we skip all health checks until the end of the transaction.
+    if !marginfi_account.get_flag(ACCOUNT_IN_RECEIVERSHIP) {
```

Comment:
I'm confused by this check. This isn't how flashloan functionality skips account health is it?

Reply:
But it doesn't really matter if borrow skips the health check while the account is in receivership because it's the liquidator that's running the transaction
- unless the account authority passes their signature as well with the liquidator's signature
- But we check account health at the end anyway

It just seems messy to have the `in_flashloan()` check inside of the `check_account_init_health` and `in_receivership()` check outside of the function.

We should be pretty confident in the setup that this can't be abused, so keeping the checks together shouldn't affect it. It's also possible to add a restriction directly on borrow that the account is not `in_receivership()`